### PR TITLE
chores: add nginx to maubot

### PR DESCRIPTION
**Pull request by amandahla**
Applicable spec: <link>

### Overview

Move NGINX to Maubot as suggested in https://github.com/canonical/maubot-operator/pull/9 by @weiiwang01 

Also remove handling exception raised by Pebble, leaving them to raise as Juju errors. The charm handle only exceptions if the user can take action. Catched it [here](https://github.com/canonical/maubot-operator/pull/7#discussion_r1764000878) by @Thanhphan1147 

### Rationale

Use only one container for Maubot.

### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The changes are compliant with [ISD054 - Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x] The documentation is generated using `src-docs`
- [ ] The documentation for charmhub is updated
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

File: maubot_rock/rockcraft.yaml [hunk 1]
```diff
@@ -16,6 +16,28 @@ environment:
...
+    stage-packages:
+      - logrotate
+      - nginx
+    plugin: nil
+    override-build: |
+      craftctl default
+      rm $CRAFT_PART_INSTALL/etc/nginx/nginx.conf
+    override-prime: |
+      craftctl default
+      mkdir run
```
Comment (arturo-seijas) [inline maubot_rock/rockcraft.yaml:40]: Why this dir?

File: maubot_rock/rockcraft.yaml [hunk 2]
```diff
@@ -16,6 +16,25 @@ environment:
   PYTHONPATH: /usr/lib/python3.12/site-packages/
 
 parts:
+  add-user:
```
Comment (jdkandersson) [inline maubot_rock/rockcraft.yaml:19]: I think the intent behind parts is more like components than actions. So perhaps rename this to `nginx-user`?

File: src/charm.py [hunk 3]
```diff
@@ -60,26 +60,18 @@ def _configure_maubot(self, container: ops.Container) -> None:
...
         ]
-        try:
-            for command in commands:
-                process = container.exec(command, combine_stderr=True)
-                process.wait()
-            config_content = str(container.pull("/data/config.yaml", encoding="utf-8").read())
-            config = yaml.safe_load(config_content)
-            config["database"] = self._get_postgresql_credentials()
-            container.push("/data/config.yaml", yaml.safe_dump(config))
-        except (ops.pebble.ExecError, ops.pebble.PathError) as exc:
```
Comment (jdkandersson) [inline src/charm.py:80]: What is the reasoning for removing the logging of errors here? Did it turn out not to be useful?

File: maubot_rock/rockcraft.yaml (hunk 2 above)
Comment (amandahla) [reply maubot_rock/rockcraft.yaml]: Good suggestion, changed.

**Review by nrobinaubertin — APPROVED** [review]

**Review by arturo-seijas — APPROVED** [review]

**Pull request merged**